sentry-android-core/src/main/java/io/sentry/android/core/internal/util/AndroidConnectionStatusProvider.java

```diff
@@ -31,104 +34,402 @@
+  private volatile @Nullable NetworkCapabilities cachedNetworkCapabilities;
+  private volatile @Nullable Network currentNetwork;
+  private volatile long lastCacheUpdateTime = 0;
+  private static final long CACHE_TTL_MS = 2 * 60 * 1000L; // 2 minutes
+
+  @SuppressLint("InlinedApi")
   public AndroidConnectionStatusProvider(
       @NotNull Context context,
-      @NotNull ILogger logger,
-      @NotNull BuildInfoProvider buildInfoProvider) {
+      @NotNull SentryOptions options,
+      @NotNull BuildInfoProvider buildInfoProvider,
+      @NotNull ICurrentDateProvider timeProvider) {
     this.context = ContextUtils.getApplicationContext(context);
-    this.logger = logger;
+    this.options = options;
     this.buildInfoProvider = buildInfoProvider;
+    this.timeProvider = timeProvider;
     this.connectionStatusObservers = new ArrayList<>();
+
+    capabilities[0] = NetworkCapabilities.NET_CAPABILITY_INTERNET;
+    if (buildInfoProvider.getSdkInfoVersion() >= Build.VERSION_CODES.M) {
+      capabilities[1] = NetworkCapabilities.NET_CAPABILITY_VALIDATED;
+    }
+
+    // Register network callback immediately for caching
+    //noinspection Convert2MethodRef
+    initThread = new Thread(() -> ensureNetworkCallbackRegistered());
+    initThread.start();
   }
 
-  @Override
-  public @NotNull ConnectionStatus getConnectionStatus() {
-    final ConnectivityManager connectivityManager = getConnectivityManager(context, logger);
-    if (connectivityManager == null) {
-      return ConnectionStatus.UNKNOWN;
+  /**
+   * Enhanced network connectivity check for Android 15. Checks for NET_CAPABILITY_INTERNET,
+   * NET_CAPABILITY_VALIDATED, and proper transport types.
+   * https://medium.com/@doronkakuli/adapting-your-network-connectivity-checks-for-android-15-a-practical-guide-2b1850619294
+   */
+  @SuppressLint("InlinedApi")
+  private boolean isNetworkEffectivelyConnected(
+      final @Nullable NetworkCapabilities networkCapabilities) {
+    if (networkCapabilities == null) {
+      return false;
+    }
+
+    // Check for general internet capability AND validated status
+    boolean hasInternetAndValidated =
+        networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET);
+    if (buildInfoProvider.getSdkInfoVersion() >= Build.VERSION_CODES.M) {
+      hasInternetAndValidated =
+          hasInternetAndValidated
+              && networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED);
     }
-    return getConnectionStatus(context, connectivityManager, logger);
-    // getActiveNetworkInfo might return null if VPN doesn't specify its
-    // underlying network
 
-    // when min. API 24, use:
-    // connectivityManager.registerDefaultNetworkCallback(...)
+    if (!hasInternetAndValidated) {
+      return false;
+    }
+
+    // Additionally, ensure it's a recognized transport type for general internet access
+    return networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)
```

Comment:
We could iterate the transports here instead
```suggestion
 for (final int transport in transports) {
   if (networkCapabilities.hasTransport(transport)) {
     return true;
   }
 }
```